src/server.js

```diff
@@ -218,6 +218,10 @@ function Server(host, port, key, cert)
             server = http.createServer(connectApp);
         }
 
+        server.on("error", function (err) {
```

Comment:
An `error` may be emitted even after the `listening` event fired and this will lead to a double invocation of the input callback to the `start()` function. I would suggest to change the code to make sure that the `start()` callback is invoked only once: without error on `listening` event received, and with error in case of an error before `listening` event is received.

Reply:
We thought about this again and came to the insight, that the error case and the "success" case are not completely disjunctive, because an error can happen even after the "listening" state was already reached and the listening state can be reached even after an "non-critical" error occurred. Therefor implemented it by using to separate callbacks for these to independent events.